### PR TITLE
reactAutocomplete

### PR DESCRIPTION
**Pull request by NasyTheSlayer**
[DEMO LINK](https://NasyTheSlayer.github.io/react_autocomplete/)

### CODE REVIEW

**Review by etojeDenys — CHANGES_REQUESTED** [review]
implement this option
image

File: src/components/Autocomplete.tsx [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    setIsUserTyping(true);
+
+    if (event.target.value === '') {
+      onSelected(null);
+    }
+  };
+
+  const handleEscape = useCallback(
+    (event: KeyboardEvent) => {
+      if (event.key === 'Escape') {
```
Comment (OleziO) [inline src/components/Autocomplete.tsx:68]: Move all keys to the enum.

**Review by OleziO — CHANGES_REQUESTED** [review]

**Review by lerastarynets — CHANGES_REQUESTED** [review]
well done in fixing prev comments, however there r still some issues,
dropdown isn't shown if a person is selested
your project
Screenshot at Nov 16 12-20-10

how it's supposed to be
Screenshot at Nov 16 12-16-18

File: src/components/Autocomplete.tsx [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+  };
+
+  const handleQueryChange = (
+    event: React.ChangeEvent<HTMLInputElement>,
+  ) => {
+    setQuery(event.target.value);
+    applyQuery(event.target.value);
+    setIsUserTyping(true);
+
+    if (event.target.value === '') {
```
Comment (VitaliyBodnar) [inline src/components/Autocomplete.tsx:71]:
```suggestion
    if (!event.target.value) {
```

File: src/components/Autocomplete.tsx [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+    >
+      <div className="dropdown-trigger">
+        <input
+          ref={inputRef}
+          type="text"
+          placeholder="Enter a part of the name"
+          className="input"
+          value={query}
+          onChange={handleQueryChange}
+          onFocus={() => setShowDrop(true)}
```
Comment (VitaliyBodnar) [inline src/components/Autocomplete.tsx:107]: It's a bad practice to create jsx anonymous function, it's better to create separate function for this
```suggestion
          onFocus={handleOnFocus}
```
Check other places

**Review by VitaliyBodnar — CHANGES_REQUESTED** [review]

File: src/components/Autocomplete.tsx [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+                  'dropdown-item',
+                  { 'has-text-danger': person.sex === 'f' },
+                  { 'has-text-link': person.sex === 'm' },
+                )}
+                onMouseDown={(event) => {
+                  event.stopPropagation();
+                  event.preventDefault();
+                  handleItemChange(event, person);
+                  setShowDrop(false);
+                }}
```
Comment (VitaliyBodnar) [inline src/components/Autocomplete.tsx:132]: Please create a separate function for onMouseDown event as it will make your code cleaner

**Review by VitaliyBodnar — CHANGES_REQUESTED** [review]

**Review by anastasiiavorobiova — APPROVED** [review]
Great work!

**Pull request closed without merging**